### ISSUE TITLE
Java Debugger Issues

### ISSUE THREAD

**Issue by JesterOrNot** [gitpod-io/gitpod#886]
**Describe the bug**
![image](https://user-images.githubusercontent.com/50386709/68050167-92df5980-fcb2-11e9-9199-159c7557a28e.png)
Java debugger is possibly broken
**To Reproduce**
Try to debug a Java app
**Expected behavior**
debug java applications
**Additional Information**
here is example repo
https://github.com/thecla1/Rubik

**Comment by svenefftinge** [comment]
I've seen this happening, too. Seems to be a race during initialization.
It only happens sporadically. Reloading the browser tab usually helps.

**Comment by alasdairg** [comment]
This has been happening to me virtually every time I start a workspace. Another symptom of this problem I have noticed is that Code Lens does not appear to have started either (No Run/Debug links appear next to a main() method in a Java class, for example). Attempting to run or debug just results in a "No delegateComandHandler for vscode.java.resolveMainMethod" popup notification.

A workaround I found was to locate the main gitpod-node process in the terminal:

"/theia/node/bin/gitpod-node ./src-gen/backend/main.js ...."

and kill it. This forces everything to restart, and usually that fixes it. I guess this has roughly the same effect as reloading the browser tab as noted above, which is simpler ...

A little annoying though!
